File: src/core/main/ui/ImageElement.java
package core.main.ui;

import core.main.VGraphics;
import core.main.structs.Vector;
import java.awt.image.BufferedImage;

public class ImageElement implements IElement{

    private BufferedImage image;
    
    public void setImage(BufferedImage i){ image = i; }

    public Vector getSize() {
        if(image==null){ return new Vector(); }
        return new Vector(image.getWidth(), image.getHeight());
    }

    public void render(VGraphics g) {
        g.drawImage(image, new Vector());
    }
}

File: src/velvet/velements/impl/ImageElement.kt
package velvet.velements.impl

import velvet.main.VGraphics
import velvet.structs.Vector
import velvet.velements.VElement
import java.awt.image.BufferedImage

class ImageElement (var image: BufferedImage? = null) : VElement {

    override val size: Vector?
        get() = image?.let(::calculateImageSize)

    fun calculateImageSize(image: BufferedImage): Vector{
        return Vector(image.width.toDouble(), image.height.toDouble())
    }

    override fun render(g: VGraphics, targetSize: Vector) {
        image?.let {
            g.save()
            g.scale(targetSize.divide(calculateImageSize(it)))
            g.drawImage(it)
            g.reset()
        }
    }
}